src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -249,4 +255,27 @@ public Section(string name, ImmutableDictionary<string, string> properties)
             public ImmutableDictionary<string, string> Properties { get; }
         }
     }
+
+    /// <summary>
+    /// Represents a combined global analyzer config.
+    /// </summary>
+    /// <remarks>
+    /// We parse all <see cref="AnalyzerConfig"/>s as individual files, according to the editorconfig spec.
+    /// 
+    /// However, when viewing the configs as an <see cref="AnalyzerConfigSet"/> if multiple files have the
+    /// <c>is_global</c> property set to <c>true</c> we combine those files and treat them as a single 
+    /// 'logical' global config file. This type represents that combined file. 
+    /// </remarks>
+    internal sealed class GlobalAnalyzerConfig
```

Comment:
I don't see any usage of this type outside of `AnalyzerConfigSet` let's move it to a nested type under there. Or, alternatively, use a tuple, since it seems like that's what this is :)

Reply:
I so nearly made this a tuple, but I can't do it. *mutters about primitive obsession*. Moved to nested.

---
In reply to: [423391173](https://github.com/dotnet/roslyn/pull/43889#discussion_r423391173) [](ancestors = 423391173)